File: peek_plugin_unified_search/plugin-module/_private/tuples/SearchResultOpenHandlerTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {unifiedSearchTuplePrefix} from "../PluginNames";


@addTupleType
export class SearchResultOpenHandlerTuple extends Tuple {
    public static readonly tupleName = unifiedSearchTuplePrefix + "SearchResultOpenHandlerTuple";

    // The key of a registered open handler
    key: string;

    // The name of the open handler
    name: string;

    // The description of the open handlers action type
    title: string;

    constructor() {
        super(SearchResultOpenHandlerTuple.tupleName)
    }
}

File: peek_plugin_unified_search/_private/tuples/SearchResultOpenHandlerTuple.py
from typing import List

from vortex.Tuple import Tuple, addTupleType, TupleField

from peek_plugin_unified_search._private.PluginNames import unifiedSearchTuplePrefix
from .SearchResultDetailTuple import SearchResultDetailTuple


@addTupleType
class SearchResultOpenHandlerTuple(Tuple):
    """ Search Result Tuple

    This tuple represents a search result
    """
    __tupleType__ = unifiedSearchTuplePrefix + 'SearchResultTuple'

    #:  The key of a registered open handler
    key: str = TupleField()

    #:  The name of the open handler
    name: str = TupleField()

    #:  The description of the open handlers action type
    title: str = TupleField()
